lib/rules/prefer-wait-for.ts

```diff
@@ -119,36 +115,44 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
       'ImportDeclaration[source.value=/testing-library/]'(
         node: TSESTree.ImportDeclaration
       ) {
-        const importedNames = node.specifiers
-          .filter(
-            specifier => isImportSpecifier(specifier) && specifier.imported
-          )
-          .map(
-            (specifier: TSESTree.ImportSpecifier) => specifier.imported.name
-          );
-
-        if (
-          importedNames.some(importedName =>
-            DEPRECATED_METHODS.includes(importedName)
-          )
-        ) {
-          importNodes.push(node);
-        }
-      },
-      'CallExpression Identifier[name=/^(wait|waitForElement|waitForDomChange)$/]'(
-        node: TSESTree.Identifier
-      ) {
-        waitNodes.push(node);
-      },
-      'Program:exit'() {
-        waitNodes.forEach(waitNode => {
-          reportWait(waitNode);
-        });
+        const deprecatedImportSpecifiers = node.specifiers.filter(
+          specifier =>
+            isImportSpecifier(specifier) &&
+            specifier.imported &&
+            DEPRECATED_METHODS.includes(specifier.imported.name)
+        );
+
+        deprecatedImportSpecifiers.forEach((importSpecifier, i) => {
+          if (i === 0) {
+            reportImport(node);
+          }
 
-        importNodes.forEach(importNode => {
-          reportImport(importNode);
+          context
+            .getDeclaredVariables(importSpecifier)
+            .forEach(variable =>
+              variable.references.forEach(reference =>
+                reportWait(reference.identifier)
+              )
+            );
```

Comment:
Is there a way we could make this cleaner? 👀

Reply:
perhaps

```ts
context
  .getDeclaredVariables(importSpecifier)
  .flatMap(variable => variable.references)
  .forEach(({ identifier })=> reportWait(identifier))
```

? But IMO it's not too much of a difference - I think this is fine

And I think flatMap does not work in node <= 10

Reply:
Indeed, we can't use [flatMap in node 10](https://node.green/#ES2019-features-Array-prototype--flat--flatMap-)

Reply:
Maybe we could make the cod cleaner by extracting the first forEach callback to its own function? It's the only thing I can think of.